wg-code-quality/wg-charter.md

```diff
@@ -0,0 +1,52 @@
+# WG Code-Quality Charter
+
+## Goals
+
+The Code-Quality Working Group has two goals:
+
+* (short-term) **focus on the improvement of code quality** and
+* (long-term) **create an inter-SIG process for identifying, agreeing on and addressing code quality issues**
+
+The long-term goal is achieved when the process is implemented and executed as required.
+
+## Definition of objectives
+
+The main objective of this effort is to improve the quality of the project codebase.
+
+With this in mind, there is a need to define a clear scope on which areas and through what means this can be accomplished.
+
+* Create and maintain code-quality guidelines, best practices and standards.
+* Assure distribution of knowledge in regard to code quality.
+  * This may include presentations, documentation, lectures, certifications, linters, etc.
+* Enforcement of selected standards.
+  * Through automation and active intervention in code reviews.
+  * Through codebase infrastructure.
+* Management:
+  * Prioritizing tasks.
+  * Focusing on selected topics.
+* Monitor and identification of code-quality issues.
+
+## Deliverables
+
+The inter-SIG process definition will be manifested in a document that defines the process itself, also how and when it is executed. 
```

Comment:
I think the deliverables should be the result of the scope points (objectives in the current naming), more or less.

I like the general wording of the device-manager WG:

> The WG will coordinate the delivery of KEPs and their implementations by the participating SIGs. Interim artifacts will include documents capturing use cases, requirements, and designs; however, all of those will eventually result in KEPs and code owned by SIGs.

We could re-word it a bit to fit better to our naming (or lack of it) and scope:

> The WG will coordinate the prioritization and delivery of design-proposals, documentation, implementations and other content that aims to improve the code quality of the project.